AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthenticationProviderTests/AuthenticationProviderSigninTests.swift

```diff
@@ -39,7 +50,784 @@ class AuthenticationProviderSigninTests: BaseAuthenticationProviderTest {
                 XCTFail("Received failure with error \(error)")
             }
         }
-        wait(for: [resultExpectation], timeout: 2)
+        wait(for: [resultExpectation], timeout: apiTimeout)
     }
 
+    /// Test a signIn with empty username
+    ///
+    /// - Given: Given an auth plugin with mocked service.
+    ///
+    /// - When:
+    ///    - I invoke signIn with empty username
+    /// - Then:
+    ///    - I should get a .validation error
+    ///
+    func testSignInWithEmptyUsername() {
+
+        let mockSigninResult = SignInResult(signInState: .signedIn)
+        mockAWSMobileClient?.signInMockResult = .success(mockSigninResult)
+        let options = AuthSignInRequest.Options()
+
+        let resultExpectation = expectation(description: "Should receive a result")
+        _ = plugin.signIn(username: "", password: "password", options: options) { result in
+            defer {
+                resultExpectation.fulfill()
+            }
+            switch result {
+            case .success(let signinResult):
+                XCTFail("Should not receive a success response \(signinResult)")
+            case .failure(let error):
+                guard case .validation = error else {
+                    XCTFail("Should receive validation error instead got \(error)")
+                    return
+                }
+            }
+        }
+        wait(for: [resultExpectation], timeout: apiTimeout)
+    }
+
+    /// Test a signIn with empty password
+    ///
+    /// - Given: Given an auth plugin with mocked service.
+    ///
+    /// - When:
+    ///    - I invoke signIn with no password
+    /// - Then:
+    ///    - I should get a valid response
+    ///
+    func testSignInWithEmptyPassword() {
+
+        let mockSigninResult = SignInResult(signInState: .signedIn)
+        mockAWSMobileClient?.signInMockResult = .success(mockSigninResult)
```

Comment:
Feels like empty password should return a service error? 

According to [this](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_SignUp.html)
It says: Length Constraints: Minimum length of 6. Maximum length of 256.

Reply:
Nope, our signIn api takes empty password, for passwordless signIn